completion/available/aliases.completion.bash

```diff
@@ -40,6 +40,24 @@ function _bash-it-component-completion-callback-on-init-aliases() {
 		line="${line#alias -- }"
 		line="${line#alias }"
 		alias_name="${line%%=*}"
+
+		# Skip aliases not added by this script that already have completion functions.
+		# This allows users to define their own alias completion functions.
+		# For aliases added by this script, we do want to replace them in case the
+		# alias getting the completion added has changed.
+		if complete -p "$alias_name" &> /dev/null; then
+			# Get the -F argument from the existing completion for this alias.
+			aliasCommandFunction=$(complete -p "$alias_name" | rev | cut -d " " -f 2 | rev)
```

Comment:
Shouldn't this variable be localized?

Also, I'm not sure if one can safely assume that the second last word of `complete -p <alias>` would be the argument of the `-F` option. There can be completion settings with `-C <cmd>` or ones without neither `-F` nor `-C`.

Reply:
You are absolutely right. Adding it as a local var in my next PR.

@convergedtarkus you said you fully tested this, right? Got any input regarding the `complete` issue raised by Koichi-san?

Reply:
I agree on making the variable local.

As for the logic, I think the logic is safe given that the scope is isolated.

This logic checks if the current alias being analyzed has an existing completion. If it does, this command tries to extract the name of the function associated with the completion. If the extracted name is anything but the namespace used for the aliases added by this script, that alias is skipped over because it has an existing completion function that was not added by this script. So the script only cares about aliases with existing completion functions added by this script.

Since this code is only looking for the completion commands added by this script, we can know they are created in a predictable manner. Anything else just ends up being skipped.

Reply:
Thank you for the explanation. I haven't carefully looked at the code around this line. Checking prefix `_${namespace}::` (where `namespace` is `alias_completion`) should usually be sufficient. You can mark this conversation as marked (I'm not the PR author or a maintainer, so I don't have access to the resolve button).